src/mcmc/gibbs.jl

```diff
@@ -433,28 +491,36 @@ function setparams_varinfo!!(model, ::Sampler, state, params::AbstractVarInfo)
 end
 
 function setparams_varinfo!!(
-    model::DynamicPPL.Model, sampler::Sampler{<:MH}, state::VarInfo, params::AbstractVarInfo
+    model::DynamicPPL.Model,
+    sampler::Sampler{<:MH},
+    state::AbstractVarInfo,
+    params::AbstractVarInfo,
 )
     # The state is already a VarInfo, so we can just return `params`, but first we need to
     # update its logprob.
-    # TODO(mhauru) Is this the right context to use?
+    # NOTE: Using `leafcontext(model.context)` here is a no-op, as it will be concatenated
+    # with `model.context` before hitting `model.f`.
     return last(DynamicPPL.evaluate!!(model, params, DynamicPPL.leafcontext(model.context)))
 end
 
 function setparams_varinfo!!(
     model::DynamicPPL.Model,
     sampler::Sampler{<:ESS},
-    state::VarInfo,
+    state::AbstractVarInfo,
     params::AbstractVarInfo,
 )
     # The state is already a VarInfo, so we can just return `params`, but first we need to
-    # update its logprob.
-    # TODO(mhauru) Is this the right context to use?
-    return last(DynamicPPL.evaluate!!(model, params, DynamicPPL.LikelihoodContext()))
+    # update its logprob. To do this, we have to call evaluate!! with the sampler, rather
+    # than just a context, because ESS is peculiar in how it uses LikelihoodContext for
+    # some variables and DefaultContext for others.
+    return last(DynamicPPL.evaluate!!(model, params, sampler))
```

Comment:
```suggestion
    return last(DynamicPPL.evaluate!!(model, params, SamplingContext(sampler))
```
just in case we remove the other `evaluate!!` later

Reply:
Done.